docs/advanced_topics/icons.rst

```diff
@@ -0,0 +1,104 @@
+.. _icons:
+
+=====
+Icons
+=====
+
+Wagtail comes with an icon set.
+The icons are used throughout the admin interface.
+
+Elements that use icons are:
+
+- :ref:`Various menu items <admin_hooks>`
+- :ref:`ModelAdmin menu items <modeladmin_menu_icon>`
+- :ref:`Streamfield blocks <basic_block_type_icon>`
+- :ref:`Rich text editor toolbar buttons <extending_the_draftail_editor>`
+- ...
+
+This document describes how to choose and add icons.
+
+Available icons and their names
+-------------------------------
+
+Enable the :ref:`styleguide <styleguide>` to view icons and their names.
```

Comment:
Note from https://github.com/wagtail/wagtail/pull/6493#discussion_r588925559 – as part of this documentation work, we could consider adding the icon names directly to the docs.